### PR TITLE
Add calendar tests

### PR DESCRIPTION
**Pull request by arturs105**
This adds tests that were missing from #1174 

Squash whenever.

**Edit:** since i'm on vacation next week, I hereby give permission to @willsb, @simonrozsival and @kornelijepetak to make all requested changes and merge this PR.

### CODE REVIEW

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/ReportsCalendarViewModelTests.cs [hunk 1]
```diff
@@ -72,6 +81,328 @@ public sealed class TheInitializeMethod : ReportsCalendarViewModelTest
                     ViewModel.Months[i].CalendarMonth.Should().Be(month);
                 }
             }
+
+            [Fact, LogIfTooSlow]
+            public async Task FillsQuickSelectShortCutlist()
```
Comment (kornelijepetak) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/ReportsCalendarViewModelTests.cs]: Shortcut is a single atomic word and should not be spelled with a capital `C`. Also, `List` should :)

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/CalendarQuickSelectShortcuts/BaseCalendarQuickSelectShortcutTests.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        where T : CalendarBaseQuickSelectShortcut
+    {
+        protected ITimeService TimeService { get; } = Substitute.For<ITimeService>();
+
+        protected BaseCalendarQuickSelectShortcutTests()
+        {
+            TimeService.CurrentDateTime.Returns(CurrentTime);
+        }
+
+        protected abstract T CreateQuickSelectShortCut();
```
Comment (kornelijepetak) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/CalendarQuickSelectShortcuts/BaseCalendarQuickSelectShortcutTests.cs]: Shortcut should not be spelled with a capital `C`.

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/CalendarQuickSelectShortcuts/BaseCalendarQuickSelectShortcutTests.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        protected abstract T TryToCreateQuickSelectShortCutWithNull();
+
+        protected abstract DateTimeOffset CurrentTime { get; }
+        protected abstract DateTime ExpectedStart { get; }
+        protected abstract DateTime ExpectedEnd { get; }
+
+        [Fact]
+        public void SetsSelectedToTrueWhenReceivesOnDateRangeChangedWithOwnDateRange()
+        {
+            var quickSelectShortCut = CreateQuickSelectShortCut();
```
Comment (kornelijepetak) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/CalendarQuickSelectShortcuts/BaseCalendarQuickSelectShortcutTests.cs]: Shortcut should not be spelled with a capital `C`. (here and everywhere below)

**Comment by arturs105** [comment]
The test was removed, because currently selecting "This year" is broken and it's fixed in #1248 along with adding a test for the fixed behaviour.

**Pull request merged**